src/rosdep2/sources_list.py

```diff
@@ -437,9 +440,30 @@ def update_sources_list(sources_list_dir=None, sources_cache_dir=None,
     """
     if sources_cache_dir is None:
         sources_cache_dir = get_sources_cache_dir()
-
+    
     sources = parse_sources_list(sources_list_dir=sources_list_dir)
     retval = []
+
+    # Obtain python version for cache index
+    version_string = sys.version[:3]
+
+    # Verify that the contents of the cache was
+    # written with a comaptible python version.
+    if os.path.exists(sources_cache_dir):
+        cache_index = os.path.join(sources_cache_dir, CACHE_INDEX)
+        try:
+            for line in open(cache_index, 'r').read().split('\n'):
+                if "# Python" in line and version_string not in line:
+                    err_str = "incompatible python version detected:\n"
+                    err_str += "cached version is " + line.split(' ')[3]
+                    err_str += " but system version is " + version_string + ".\n"
+                    err_str += "Please remove the current cache and re-run rosdep update."
+                    raise InvalidData("%s"%(err_str), origin='<string>')
+        except (OSError, IOError) as e:
+            err_str = "cache index file does not exist but cache directory does.\n"
+            err_str += "remove cache directory and re-run rosdep update."
```

Comment:
It would be great if this error message contained the path to the cache directory.